lib/amp/plugin.js

```diff
@@ -1,7 +1,41 @@
 import './middleware'
 
+function pick(a, b) {
+  return a === undefined ? b : a
+}
+
 export default function ({ route, req }, inject) {
-  const isAMP = Boolean(route.query.amp)
+  if (!route.matched[0]) {
+    return
+  }
+
+  const { options } = route.matched[0].components.default
+
+  const ampOption = pick(
+    options.amp,
+    pick(
+      route.meta.amp,
+      'hybrid'
+    )
+  )
+
+  let isAMP = false
+
+  switch (ampOption) {
```

Comment:
As the variable name is also injected to the context, what do you think to call it something like `ampMode`?

Reply:
Sounds good to me, `ampMode` is more meaningful